aten/src/ATen/native/cpu/ScatterGatherKernel.cpp

```diff
@@ -176,10 +176,11 @@ void gather_cpu_kernel(Tensor& result, const Tensor& self, int64_t dim, const Te
     ) {
       for (int64_t i = 0; i < index_dim_size; ++i) {
         int64_t idx_dim = index_data[i * index_dim_stride];
+        // we are not putting idx_dim in the error message because it disables
+        // loop optimization in some compilers
```

Comment:
Could it be CPU-specific? Not all stations in my experiments showed performance regression with `clang7`.

Reply:
It could also be cpu-specific, although when I pulled ScatterGather.cpp out of cpu folder to disable compilation with avx the regression was still present.

Reply:
OK, I tried making a separate checking function, and checking on if, and the regression was still there. Now I wonder could it potentially affect the performance of `cpu_index_kernel`, the `get` function specifically?

Reply:
It is affected, but the best I can do for it is some restructuring and a constant error message, and I don't know if it's worth it
```
   for (int j = 0; j < num_indexers; j++) {
      int64_t value = *(int64_t*)&indexers[j][idx * indexer_strides[j]];
      int64_t size = original_sizes[j];
      if (value < 0) {
        value += size;
      }
      offset += value * original_strides[j];
      if (value < 0 || value >= size) {
        AT_INDEX_ERROR("index is out of bounds");
      }
    }
```
The benchmarking I used for index is
```
dim1ind = torch.arange(N).view(1,N).expand(M,N)
start = time.time()
for i in range(nrep):
    aa=input_one[index, dim1ind]
end = time.time()
print("index", (end - start)/nrep)
```
the same operation as gather does, essentially.